flowcell/views.py

```diff
@@ -356,6 +356,15 @@ def create_row(lane, record):
     @list_route(methods=['get'])
     def analysis_list(self, request):
         flowcell_id = request.query_params.get('flowcell_id', '')
+
+        # Don't throw an error if the flow cell ID is missing
```

Comment:
this all can be replaced with just one line :)
```
from django.shortcuts import get_object_or_404

...

flowcell = get_object_or_404(Flowcell, flowcell_id=flowcell_id)
```

Reply:
Ah, figures :)